SDWebImage/SDImageCache.h

```diff
@@ -143,7 +144,36 @@ typedef void(^SDWebImageCalculateSizeBlock)(NSUInteger fileCount, NSUInteger tot
  * @param imageData  The image data to store
  * @param key        The unique image cache key, usually it's image absolute URL
  */
-- (void)storeImageDataToDisk:(nullable NSData *)imageData forKey:(nullable NSString *)key;
+- (void)storeImageDataToDisk:(nullable NSData *)imageData
+                      forKey:(nullable NSString *)key;
+
+/**
+ * Synchronously store image NSData into disk cache at the given key.
+ *
+ * @warning This method is synchronous, make sure to call it from the ioQueue
+ *
+ * @param imageData  The image data to store
+ * @param key        The unique image cache key, usually it's image absolute URL
+ * @param errorPtr   NSError pointer. If error occurs then (*errorPtr) != nil.
+ */
+- (BOOL)storeImageDataToDisk:(nullable NSData *)imageData
+                      forKey:(nullable NSString *)key
+                       error:(NSError * _Nullable * _Nullable)errorPtr;
+
+/**
+ * Synchronously store image NSData into disk cache at the given key.
+ *
+ * @warning This method is synchronous, make sure to call it from the ioQueue
+ *
+ * @param imageData   The image data to store
+ * @param key         The unique image cache key, usually it's image absolute URL
+ * @param fileManager The file manager for storing image. If nil, then will be used local object.
+ * @param errorPtr    NSError pointer. If error occurs then (*errorPtr) != nil.
+ */
+- (BOOL)storeImageDataToDisk:(nullable NSData *)imageData
+                      forKey:(nullable NSString *)key
+                 fileManager:(nullable NSFileManager *)fileManager
+                       error:(NSError * _Nullable * _Nullable)errorPtr;
```

Comment:
I don't think we need all those methods, we can just refactor the old `storeImageDataToDisk:forKey:` into `storeImageDataToDisk:forKey:error:`.
The proper way of using dependency injection for the `NSFileManager` is creating a new initializer for `SDImageCache` that allows passing the file manager as well.